src/typeahead/typeahead.js

```diff
@@ -127,10 +127,12 @@ angular.module('ui.bootstrap.typeahead', ['ui.bootstrap.debounce', 'ui.bootstrap
       inputsContainer = angular.element('<div></div>');
       inputsContainer.css('position', 'relative');
       element.after(inputsContainer);
-      hintInputElem = element.clone();
-      hintInputElem.attr('placeholder', '');
+      hintInputElem = angular.element('<input></input>');
       hintInputElem.attr('tabindex', '-1');
-      hintInputElem.val('');
+      hintInputElem.attr('class', element.attr('class'));
+      hintInputElem.attr('style', element.attr('style'));
+      hintInputElem.attr('ng-class', element.attr('ng-class'));
+      hintInputElem.attr('ng-disabled', element.attr('ng-disabled'));
```

Comment:
This also needs to be done for the regular `disabled` attribute & use `.prop` in addition to this.